### ISSUE TITLE
Crashing with latest compose BOM 2023.03.00

### ISSUE THREAD

**Issue by ColtonIdle** [oleksandrbalan/modalsheet#22]
```
java.lang.NoSuchMethodError: No virtual method getProgress()Landroidx/compose/material/SwipeProgress; in class Landroidx/compose/material/ModalBottomSheetState; or its super classes (declaration of 'androidx.compose.material.ModalBottomSheetState' appears in /data/app/~~c68PNDi44D--xILc_StEgg==/com.rollertoaster.app-as==/base.apk)
at eu.wewox.modalsheet.ModalSheetKt.ModalSheet-BzaUkTc(ModalSheet.kt:133)
at eu.wewox.modalsheet.ModalSheetKt.ModalSheet-BzaUkTc(ModalSheet.kt:61)
at com.MyModalSheetKt.MyModalSheet
```

**Comment by wooodenleg** [comment]
Thanks for the report, Colton! 

@oleksandrbalan we've discussed this change before but I think we kinda slept on it and now the stable version of 1.4 is out :D 
The new version of sheet state has (at least in M3 but should be the same in M2) `sheetState.requireOffset()` function. Maybe it can be used for the same thing as progress before. But if I remember correctly the progress was used only for mitigating some bug/instability and maybe that's already fixed here 🤔

**Comment by oleksandrbalan** [comment]
Yea, I have been experimenting with it and yesterday I pushed an update to the separate branch and released 0.4.0-SNAPSHOT from it. 

On Slack Colton said he will test it 🙏 and if everything works fine we could release a 0.4.0 with a newest compose version.

@wooodenleg I will soon create a PR with it 👌

**Comment by oleksandrbalan** [comment]
The brand new 0.4.0 version is released with BOM 2023.03.00 🎉 

Thx everyone for testing and review 🙇
